### PR TITLE
Budget duration made optional

### PR DESCRIPTION
**Pull request by KvelaGorrrrnio**
- Segment budget duration is marked as optional and converts `null` values to `undefined`.

### CODE REVIEW

File: src/model/entities/segment.ts [hunk 1]
```diff
@@ -54,6 +54,9 @@ export class Segment {
   }
 
   public setAsNext(): void {
+    if (!this.isSegmentOnAir) {
+      this.reset()
+    }
     this.isSegmentNext = true
     if (!this.isSegmentOnAir) {
       this.reset()
```
Comment (LindvedKrvang) [inline src/model/entities/segment.ts:62]: I think you should take a closer look at this code ;)

File: src/data-access/repositories/mongo/mongo-entity-converter.ts [hunk 2]
```diff
@@ -148,7 +148,7 @@ export class MongoEntityConverter {
       isOnAir: false,
       isNext: false,
       parts: [],
-      budgetDuration: mongoSegment.budgetDuration,
+      budgetDuration: mongoSegment.budgetDuration ?? undefined, // Ensure that null values are stripped
```
Comment (LindvedKrvang) [inline src/data-access/repositories/mongo/mongo-entity-converter.ts:151]: `undefined` values aren't much better than `null` values...

File: src/model/entities/segment.ts (hunk 1 above)
Comment (LindvedKrvang) [reply src/model/entities/segment.ts:62]: One could even say you should do a double take :D

**Review by andr9528 — APPROVED** [review]
As long as the woopsy Lindved commented on is fixed, i am happy with it.

File: src/model/entities/segment.ts (hunk 1 above)
Comment (KvelaGorrrrnio) [reply src/model/entities/segment.ts]: The reset below is from develop and not this branch. I have merged develop into the branch and removed the double reset.

File: src/data-access/repositories/mongo/mongo-entity-converter.ts (hunk 2 above)
Comment (KvelaGorrrrnio) [reply src/data-access/repositories/mongo/mongo-entity-converter.ts:163]: No, but they are better than nullish values :)

**Review by RasmusAlbrektsen — APPROVED** [review]

**Pull request merged**